### PR TITLE
[EntityGenerator] add the support of ApiPlatform

### PR DESCRIPTION
**Pull request by Simperfit**
I would like to add the support of ApiPlatform in maker, since api-platform is already in the recipes of flex.
dsc_0003
(PRception)

### CODE REVIEW

File: src/Command/MakeEntityCommand.php [hunk 1]
```diff
@@ -46,13 +49,14 @@ protected function getParameters(): array
             'entity_class_name' => $entityClassName,
             'entity_alias' => $entityAlias,
             'repository_class_name' => $repositoryClassName,
+            'api_platform' => $this->input->hasOption('with-api-platform') ? 'Api' : ''
```
Comment (fabpot) [inline src/Command/MakeEntityCommand.php]: I'd would use --with-api instead

Comment (Simperfit) [reply src/Command/MakeEntityCommand.php]: Yeah feels better, you are right.

**Comment by javiereguiluz** [comment]
Not sure about this. ApiPlatform is popular but not standard or ubiquitous. Besides, ApiPlatform is so nice that it doesn't require much boilerplate: this maker would only add this to the generated entity -> `@ApiResource`

File: src/Command/MakeEntityCommand.php [hunk 2]
```diff
@@ -72,6 +75,12 @@ protected function writeNextStepsMessage(array $params, ConsoleStyle $io)
 
     protected function configureDependencies(DependencyBuilder $dependencies)
     {
+        if ($this->input->hasOption('with-api') && true === $this->input->getOption('with-api')) {
```
Comment (stof) [inline src/Command/MakeEntityCommand.php]: `hasOption` is useless. This command always has this option in its definition

**Comment by weaverryan** [comment]
Yep, I think I'm 👎 on this too. I don't think we should add another question to the generator. And since the annotation is pretty easy.... :)

**Pull request closed without merging**